### PR TITLE
Cmd validation

### PR DESCRIPTION
**Pull request by ambercs**
#42
https://integralsoftwareinc.visualstudio.com/IntegralTimeTracker/_workitems/edit/42

Cannot delete entry older than 48 hours.

### CODE REVIEW

File: test/TimeTracker.Library.Test/Services/TimeEntryServiceTest.cs [hunk 1]
```diff
@@ -3,30 +3,31 @@
...
         private readonly Guid userId = Guid.NewGuid();
-        
+
         private readonly TimeTrackerDbContext database;
         private readonly TimeEntryService timeEntryService;
 
-        public TimeEntryServiceTest(InMemoryDatabaseWithProjectsAndUsers inMemoryDatabase)
+        public TimeEntryServiceTest()
         {
+            var inMemoryDatabase = new InMemoryDatabaseWithProjectsAndUsers();
```
Comment (hans-integral) [inline test/TimeTracker.Library.Test/Services/TimeEntryServiceTest.cs]: I didn't do that in another test class, but just thought about this:
the InMemory... class needs to be disposed after each usage. (for releasing resources efficiently)

What to do:
make this test-class implement IDisposable and implement the Dispose method to call something like this:
```C#
if (inMemoryDatabase != null) {
    inMemoryDatabase.Dispose();
}
```

Comment (hans-integral) [inline test/TimeTracker.Library.Test/Services/TimeEntryServiceTest.cs]: I didn't do that in another test class, but just thought about this:
the InMemory... class needs to be disposed after each usage. (for releasing resources efficiently)

What to do:
make this test-class implement IDisposable (also, remove the IClassFixture) and implement the Dispose method to call something like this:
```C#
if (inMemoryDatabase != null) {
    inMemoryDatabase.Dispose();
}
```

Comment (hans-integral) [inline test/TimeTracker.Library.Test/Services/TimeEntryServiceTest.cs]: I didn't do that in another test class, but just thought about this:
the InMemory... class needs to be disposed after each usage. (for releasing resources efficiently)

What to do:
make this test-class implement IDisposable (also, remove the IClassFixture) and implement the Dispose method to call something like this:
```C#
if (inMemoryDatabase != null) {
    inMemoryDatabase.Dispose();
}
```

File: src/TimeTracker.Library/Services/TimeEntryService.cs [hunk 2]
```diff
@@ -63,11 +62,22 @@ public async Task<Guid> CreateBillableTimeEntry(DateTime date, double hours, int
...
-            var timeEntries = await db.TimeEntries.Where(x => x.UserId == userId && x.Date >= date.Date).ToListAsync();
+            var cutOffDate = DateTime.UtcNow.Date.AddHours(-48);
+            if (date < cutOffDate)
+            {
+                throw new Exception("Entries older than 48 hours cannot be deleted.");
+            }
+            
+            var timeEntries = await db.TimeEntries.Where(x => x.UserId == userId && 
+                                                              x.Date.Date == date.Date.Date && 
+                                                              x.Date >= cutOffDate).ToListAsync();
```
Comment (hans-integral) [inline src/TimeTracker.Library/Services/TimeEntryService.cs]: Do you want to floor this to just the .Date part as well?

Comment (ambercs) [reply src/TimeTracker.Library/Services/TimeEntryService.cs]: ... `.Date` floors the time and leaves the day

File: test/TimeTracker.Library.Test/Services/TimeEntryServiceTest.cs (hunk 1 above)
Comment (JamesReate) [reply test/TimeTracker.Library.Test/Services/TimeEntryServiceTest.cs]: @hans-integral I thought we had lost you as a contributor 😛
Gonna help out here to expedite, added that Dispose method, although using the newish `?.` null check  operator

**Pull request merged**